### ISSUE TITLE
Micro SD card not detected in Linux

### ISSUE THREAD

**Issue by Asgaror** [skycoin/skyflash#27]
## Expected Behavior
Micro SD card gets detected and is available for selection in step 4. 'Flash the images'

## Actual Behavior
Does not detect micro SD card. It's not relevant whether the micro SD card was inserted before or after the program was started and the images were built. Unmounting the micro SD card brings no solution to this issue, it still can't be selected for the flashing procedure. The file explorer or other flashing tools like balenaEtcher detect the micro SD card.

![image](https://user-images.githubusercontent.com/32180146/59047953-31db6300-8885-11e9-860b-80f3c15ce336.png)

## Steps to Reproduce the Problem

1. Start program without micro SD card inserted
2. Download/select local image
3. Built images
4. Program fails to detect micro SD card 
or
1. Start program with micro SD card inserted
2. Download/select local image
3. Built images
4. Program fails to detect micro SD card 

## Specifications

  - Version: `None`, no release on current [develop branch](https://github.com/skycoin/skyflash/tree/develop)
  - Platform: 
```NAME="Ubuntu"
VERSION="18.04.2 LTS (Bionic Beaver)"
ID=ubuntu
ID_LIKE=debian
PRETTY_NAME="Ubuntu 18.04.2 LTS"
VERSION_ID="18.04"
```

## Possible Solution
Maybe related to missing [privilges](https://github.com/skycoin/skyflash/blob/develop/USER_MANUAL.md#why-skyflash-ask-for-credentials-or-privileged-operations) as I was never queried to grant Skyflash any admin priviliges although running `skyflash-gui.py` as admin doesn't help either.

**Comment by stdevPavelmc** [comment]
Thanks for reporting, 

I'm using the same Ubuntu OS as main dev environment, please publish for me _(use [pastebin](https://pastebin.com) or a gists for example)_ the skyflash.log file located on $HOME/Skyflash/

What type of  sd cards reader are you using? a USB one or a built in on a laptop/PC?

Also you can list the disks drives & devices detected by your system with the SD card inserted for me, the following commands in console (cut & paste) will as for admin privileges and create a file named disks.txt on your home folder with the info I need. Please attach publish it for me to diagnose.

```sh
sudo fdisk -l > ~/disks.txt
sudo mount >> ~/disks.txt
sudo chown $USER:$USER ~/disks.txt
lspci >> ~/disks.txt
lsusb >> ~/disks.txt
```
That will allow me to debug the issue with more confidence & detail.
Usually in linux the card detection does not need the admin privilege to work, 

If you are concerned with privacy issue and are not eager to publish that info you can send it to me via telegram in a private message mentioning the Issue link, my user is @pavelmc

Thanks in advance.

**Comment by Asgaror** [comment]
@stdevPavelmc 
Sent you the `skyflash.log` file and the details on Telegram.

**Comment by stdevPavelmc** [comment]
After some analysis I think I have a flaw in the detection algorithm, working on it now...

**Comment by stdevPavelmc** [comment]
Closed in PR #39

**Comment by stdevPavelmc** [comment]
I'm reopening this issue as the problem remains, Investigation in progress... stay tuned...

**Comment by stdevPavelmc** [comment]
Latest improved code uploaded, working on my test setup (Ubuntu 18.04 & Debian 9.x)

If you want help us to fix this please insert a uSD card in your linux PC and run the attached script, share the results on the telegram Skyflash Beta Testing group.

[linuxdetect.zip](https://github.com/skycoin/skyflash/files/3288942/linuxdetect.zip)

**Comment by stdevPavelmc** [comment]
Closed via PR #43
